src/assets/TokenRatesController.ts

```diff
@@ -72,8 +79,24 @@ export class TokenRatesController extends BaseController<
 
   private tokenList: Token[] = [];
 
-  private getPricingURL(query: string) {
-    return `https://api.coingecko.com/api/v3/simple/token_price/ethereum?${query}`;
+  private getPricingURL(chainSlugIdentifier: string, query: string) {
+    return `https://api.coingecko.com/api/v3/simple/token_price/${chainSlugIdentifier}?${query}`;
+  }
+
+  private async updateChainSlugIdentifier(chainId: string) {
+    const platforms: [
+      { id: string; chain_identifier: number | null },
+    ] = await handleFetch('https://api.coingecko.com/api/v3/asset_platforms');
+    const chain = platforms.find(
+      ({ chain_identifier }) =>
+        chain_identifier !== null && String(chain_identifier) === chainId,
+    );
+    if (chain?.id) {
+      this.update({ chainSlugIdentifier: chain.id });
+    } else {
+      this.update({ chainSlugIdentifier: '' });
+    }
```

Comment:
Should this be wrapped in a `try catch` and on the `catch` block have the same case `this.update({ chainSlugIdentifier: '' });`?